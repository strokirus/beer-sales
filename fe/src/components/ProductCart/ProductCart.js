import React from 'react';
import { object, func } from 'prop-types';

import {
  Header,
} from '../Product/styles';

import {
  ItemCart,
  SectionBody,
  Commands,
} from './styles';

/**
 * Component to Product
 * @param item Object with product info
 * @param addCart Function to Add Product in Cart
*/
const ProductCart = ({ item, addCart, removeCart }) => {
  const {
    id,
    name,
    qtd,
    formatPrice,
    total,
  } = item;

  return (
    <ItemCart
      className="item-cart-product"
      key={id}
      alt={name}
      title={name}
    >
      <Header>
        <h1>{name}</h1>
      </Header>

      <SectionBody>
        <p>{`Qtd: ${qtd}`}</p>
        <p>{`${formatPrice}/un`}</p>
      </SectionBody>
      <Commands>
        <button
          className="remove-qtd"
          type="button"
          onClick={(a) => removeCart(item, a)}
        >
          -
        </button>
        <button
          className="add-qtd"
          type="button"
          onClick={(a) => addCart(item, a)}
        >
          +
        </button>
        <p>{`${total}`}</p>
      </Commands>
    </ItemCart>
  );
};

ProductCart.propTypes = {
  item: object,
  addCart: func,
  removeCart: func,
};

ProductCart.defaultProps = {
  item: { },
  addCart: () => {},
  removeCart: () => {},
};

export default ProductCart;
