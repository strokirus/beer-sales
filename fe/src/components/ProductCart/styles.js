import styled from 'styled-components';

export const ItemCart = styled.section`
    box-shadow: rgba(0,0,0,0.15) 0px 2px 4px 0px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 5px;
    text-align: center;
    margin: 10px 5px;
    padding-bottom: 10px;
`;

export const SectionBody = styled.section`
    display: grid;
    grid-template-columns: 50% 50%;
    margin-bottom: 10px;
`;

export const Commands = styled.section`
    display: grid;
    grid-template-columns: 25% 25% 50%;
    margin-bottom: 10px;
    text-align: center;

    > button {
        margin: 0 auto;
    }
`;


export default {
  ItemCart,
  SectionBody,
  Commands,
};
