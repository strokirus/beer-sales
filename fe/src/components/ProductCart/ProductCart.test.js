import React from 'react';
import renderer from 'react-test-renderer';
import {
  mount,
  configure,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ProductCart from './ProductCart';

configure({ adapter: new Adapter() });

const props = {
  item: {
    id: '9OyQXs',
    name: 'California Black Beer',
    qtd: 2,
    formatPrice: '€22,71',
    total: '€22,71',
  },
  addCart: jest.fn(),
  removeCart: jest.fn(),
};

describe('(Component) ProductCart', () => {
  it('should render an ProductCart', () => {
    const tree = renderer.create(<ProductCart {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call addCart event', () => {
    const onAddCartEventFn = jest.fn();

    const wrapper = mount(
      <ProductCart
        {...props}
        addCart={onAddCartEventFn}
      />,
    );
    expect(onAddCartEventFn).toHaveBeenCalledTimes(0);

    wrapper
      .find('.add-qtd')
      .first()
      .simulate('click');

    expect(onAddCartEventFn).toHaveBeenCalledTimes(1);
  });

  it('should call removeCart event', () => {
    const onRemoveCartEventFn = jest.fn();

    const wrapper = mount(
      <ProductCart
        {...props}
        removeCart={onRemoveCartEventFn}
      />,
    );
    expect(onRemoveCartEventFn).toHaveBeenCalledTimes(0);

    wrapper
      .find('.remove-qtd')
      .first()
      .simulate('click');

    expect(onRemoveCartEventFn).toHaveBeenCalledTimes(1);
  });
});
