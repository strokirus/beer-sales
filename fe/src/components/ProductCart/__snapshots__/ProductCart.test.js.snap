// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ProductCart should render an ProductCart 1`] = `
<section
  alt="California Black Beer"
  className="sc-AxgMl cWwlFt item-cart-product"
  title="California Black Beer"
>
  <header
    className="sc-AxiKw ehIHUN"
  >
    <h1>
      California Black Beer
    </h1>
  </header>
  <section
    className="sc-AxheI fYjabA"
  >
    <p>
      Qtd: 2
    </p>
    <p>
      €22,71/un
    </p>
  </section>
  <section
    className="sc-Axmtr iPBpgQ"
  >
    <button
      className="remove-qtd"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="add-qtd"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <p>
      €22,71
    </p>
  </section>
</section>
`;
